src/features/registry/middleware.ts

```diff
@@ -16,9 +16,13 @@ export const basicApiKeyAuthenticate: RequestHandler = async (
 		req.params.apikey = creds.pass;
 	}
 	try {
-		await retrieveAPIKey(req, undefined);
-		next();
+		const apiKey = await retrieveAPIKey(req, undefined);
+		if (apiKey != null) {
+			next();
+		} else {
+			res.status(401).end();
+		}
 	} catch (err) {
-		next(err);
```

Comment:
Isn't there an error handling middleware to handle this error already?

Reply:
Likely, but I chose to handle it here explicitly like we do on the other auth-related middleware.

Reply:
I'd rather have the shared code handle it if such exists so that it can be handled correctly for all cases, eg this should at a minimum have a `handleHttpErrors` call, and should probably be cautious about directly responding with the error message - if we don't know the error type is one that is safe to return to the user then it could leak anything, eg if somehow a db password got into the error message due to it being an error connecting to the db or something